memory/memory.ml

```diff
@@ -131,17 +144,30 @@ type memory_config = {
   xen_max_mib : int64;
   shadow_mib : int64;
   required_host_free_mib : int64;
+  overhead_mib : int64;
 }
 
 module Memory_model (D : MEMORY_MODEL_DATA) = struct
 
-  let build_max_mib static_max_mib video_mib = static_max_mib --- (Int64.of_int video_mib)
+  let build_max_mib static_max_mib video_mib =
+    static_max_mib ---
+    (Int64.of_int video_mib) +++
+    (D.shim_mib static_max_mib)
 
-  let build_start_mib target_mib video_mib = target_mib --- (Int64.of_int video_mib)
+  let build_start_mib static_max_mib target_mib video_mib =
+    if D.can_start_ballooned_down then
+      target_mib ---
+      (Int64.of_int video_mib) +++
+      (D.shim_mib target_mib)
+    else
+      build_max_mib static_max_mib video_mib
 
   let xen_max_offset_mib = D.extra_internal_mib
 
-  let xen_max_mib target_mib = target_mib +++ xen_max_offset_mib
+  let xen_max_mib static_max_mib =
+    static_max_mib +++
+    xen_max_offset_mib +++
+    (D.shim_mib static_max_mib)
```

Comment:
There is also this function https://github.com/xapi-project/squeezed/blob/master/src/squeeze_xen.ml#L68-L71 in squeezed:
```
  let maxmem_of_dominfo di =
    let memory_max_kib = Xenctrl.pages_to_kib (Int64.of_nativeint di.Xenctrl.max_memory_pages) in
    (* Misc other stuff appears in max_memory_pages *)
max 0L (memory_max_kib -* (xen_max_offset_kib di.Xenctrl.hvm_guest))
```

Will it need to be updated, or replaced to use some function from memory.ml?